pkg/syncer/pvcsi_fullsync.go

```diff
@@ -158,11 +161,37 @@ func PvcsiFullSync(ctx context.Context, metadataSyncer *metadataSyncInformer) er
 		log.Errorf("FullSync: Failed to set Guest Cluster labels on SupervisorPVC. Err: %v", err)
 		return err
 	}
-
 	log.Infof("FullSync: End")
 	return nil
 }
 
+// cleanUpCnsVolumeMetadata deletes the cnsvolumemetadata created on legacy kubernetes releases,
+// which are left unused as customer have migrated to non-legacy kubernetes releases
+func cleanUpCnsVolumeMetadata(ctx context.Context, metadataSyncer *metadataSyncInformer,
+	cnsVolumeMetadataList *cnsvolumemetadatav1alpha1.CnsVolumeMetadataList) {
+	log := logger.GetLogger(ctx)
+	log.Info("delete the cnsvolumemetadata created on legacy kubernetes releases")
+	toDeleteCnsVolumeMetadataList := cnsvolumemetadatav1alpha1.CnsVolumeMetadataList{}
+	for _, object := range cnsVolumeMetadataList.Items {
+		if object.ObjectMeta.OwnerReferences != nil {
+			for _, ownerRef := range object.ObjectMeta.OwnerReferences {
+				if ownerRef.APIVersion != apiVersionv1beta1 {
+					log.Debugf("duplicate cnsvolumemetadata %s from namespace %s is marked for deletion",
```

Comment:
It's better to check if there is another cnsvolumemetadata cr instance with the same name, and then make the decision to add the cr to toDeleteCnsVolumeMetadataList.

Reply:
as discussed ofline there will not be cnsvolumemetadata CRs with same name, but we will add check on cnsvolumemetadata. CR's Spec.EntityName+Spec.Namespace